rundeckapp/grails-app/controllers/rundeck/controllers/MenuController.groovy

```diff
@@ -2912,6 +2949,12 @@ class MenuController extends ControllerBase implements ApplicationContextAware{
      * API: /api/2/project/NAME/jobs, version 2
      */
     def apiJobsListv2 (ScheduledExecutionQuery query) {
+
+        if (!apiPaginationEnabled){
+            query.max = null
+            query.offset = null
+        }
+
```

Comment:
Since job pagination is a new "feature" it needs to only apply by default in a new API version.  You will need to increment the API version, and if the version in the request >= that version, enable it by default, otherwise it should be disabled.

Also, the job list API should then support appropriate pagination data, such as including the `offset` and `max` in the response data (both XML and JSON)

Reply:
Actually this is just for keeping previous behavior, you could ask the api for number of jobs...

This works perfectly in 3.4.9:
`curl "http://localhost:8080/api/40/project/P/jobs?max=2&authtoken=dSQvmNJG7UPNfd1LggQ5hxSXnPoH4PSp" | xml_pp -l`

The only problem was that when setting gui pagination properties it affected api responses...

Reply:
ok 👍